components-java-sdk/src/main/java/io/dapr/components/domain/bindings/ReadRequest.java

```diff
@@ -26,12 +26,13 @@
  */
 public record ReadRequest(ByteString responseData, String messageId, Optional<String> responseErrorMessage) {
   /**
-   * Conversion constructor.
+   * Conversion method.
    *
    * @param other The Protocol Buffer representation of a ReadRequest.
+   * @return The provided protocol buffer object converted into the local domain.
    */
-  public ReadRequest(dapr.proto.components.v1.Bindings.ReadRequest other) {
-    this(other.getResponseData(),
+  public static ReadRequest fromProto(dapr.proto.components.v1.Bindings.ReadRequest other) {
```

Comment:
I thought about using constructors for conversions, too, but found static methods more useful in being able to pass around as lambdas.  (If only C# allowed passing constructors as lambdas--that seems like something its compiler magic should be able to do.)

Reply:
You can in Java I think a static method documents better what is going on here.